### PR TITLE
LUCENE-10028: Add git pre-commit hook that runs precommit task.

### PR DESCRIPTION
**Pull request by mocobeta**
This adds git pre-commit hook for Lucene precommit check.
See also: https://issues.apache.org/jira/browse/LUCENE-10028

(Usage)

Install pre-commit hook (opt-in)
```
./gradlew precommitHook
```

After installing, precommit will be automatically performed before every commit.

```
git commit -m "commit attempt that doesn't pass spotless check"
......
> Task :lucene:luke:spotlessJavaCheck FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':lucene:luke:spotlessJavaCheck'.
> The following files had format violations:
......
// commit fails.
```

### CODE REVIEW

File: dev-tools/git/hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/sh
+#
+# Run precommit check
+#
+
+## TODO: Windows support
```
Comment (dweiss) [inline dev-tools/git/hooks/pre-commit:6]: I think the precommitHook task could detect the OS it's running on and copy an appropriate implementation (or fail if there is none available).

**Review by dweiss — COMMENTED** [review]
I think most folks use github pull request checks as a shortcut for this these days. A precommit hook is sometimes too much of an overhead when you're working on something in smaller steps, then reshape the commit structure (flatten, combine commits, etc.).

I don't have anything against it, just saying I wouldn't find this workflow practical.

**Comment by mocobeta** [comment]
It surely won't fit everyone's (especially git experts') use-cases, and it's a local setup anyway; devs should be able to set up `pre-commit` or `pre-push` hook without the help of Gradle if they'd like. I threw in this since I would like to make sure all linters are locally run before pushing changes to the remote repo (and I almost always forget that) though, I don't intend to force or recommend others to do so.

**Comment by mocobeta** [comment]
It's a local setup anyway. ("pre-push" hook seems to fit my usecase.)

**Pull request closed without merging**